pyunlocbox/solvers.py

```diff
@@ -780,14 +798,27 @@ def __init__(self, L=None, Lt=None, d0=None, *args, **kwargs):
     def _pre(self, functions, x0):
         # Dual variable.
         if self.d0 is None:
-            self.dual_sol = self.L(x0)
+            # The copy is necessary in case `L = lambda x: x`.
+            self.dual_sol = self.L(np.asarray(x0).copy())
         else:
             self.dual_sol = self.d0
 
     def _post(self):
         self.d0 = None
         del self.dual_sol
 
+    def objective(self, x):
```

Comment:
Could we do as for the other methods, i.e., an `objective()` method with the documentation, that calls an `_objective()` implementation, which can be overloaded?

Reply:
Done

Reply:
Thanks!